web/src/main/java/com/wilma/web/controller/web/portal/PartnerPortalController.java

```diff
@@ -28,26 +46,146 @@ public String dashboard(Model model) {
+                "menuElements", UserConfiguration.partnerMenuElements,
+                "availableCategories", categoryService.findAll(),
+                "availableTags", tagService.findAll(),
+                "contentType", type,
+                "post", new PostDTO(),
+                "reply", new ReplyDTO()));
+        if(postId != null)
+            model.addAttribute("postId", postId);
+        return "/partner/forum/forum-content";
+    }
+
+    @PostMapping("/reply-to-post")
+    public RedirectView replyToPost(@ModelAttribute ReplyDTO replyDTO, @RequestParam Integer postId, Model model){
+        var reply = forumService.addReplyFromDTO(replyDTO);
+        var category = reply.getPost().getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.partnerMenuElements,
+                "postId", postId,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category),
+                "reply", replyDTO));
+
+        log.info("Reply added: "+ reply);
+        return new RedirectView("/partner/forum-thread?category="+ category);
+    }
+
+    @PostMapping("/create-post")
+    public RedirectView createPost(@ModelAttribute PostDTO postDTO, Model model){
+        var newPost = forumService.addPostFromDTO(postDTO);
+        var categoryName = newPost.getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.partnerMenuElements,
+                "availableCategories", categoryService.findAll(),
+                "availableTags", tagService.findAll(),
+                "categoryName", categoryName,
+                "postsByCategory", forumService.getPostByCategoryName(categoryName),
+                "repliesForPosts", forumService.getPostRepliesByCategory(categoryName),
+                "post", postDTO));
+
+        log.info("Post created from DTO: "+ newPost);
+        return new RedirectView("/partner/forum");
+    }
+
+    @GetMapping("/forum-thread")
+    public String forumThread(@RequestParam String category, Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.partnerMenuElements,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return "/partner/forum/forum-thread";
+
+    }
+    //endregion
+
+    //region Expressions of interest
+    @GetMapping("/expressions-of-interest")
+    public String expressionsOfInterest(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Expressions Of Interest",
```

Comment:
Menu item not being selected
```suggestion
"currentPage", "expressions-of-interest",
```